zipkin-server/src/main/java/zipkin2/server/internal/throttle/ThrottledStorageComponent.java

```diff
@@ -102,18 +103,20 @@ public ThrottledStorageComponent(StorageComponent delegate, MeterRegistry regist
   }
 
   static final class ThrottledSpanConsumer implements SpanConsumer {
-    final SpanConsumer delegate;
-    final Limiter<Void> limiter;
     final ExecutorService executor;
+    final Limiter<Void> limiter;
+    final Predicate<Throwable> isOverCapacity;
+    final SpanConsumer delegate;
 
-    ThrottledSpanConsumer(SpanConsumer delegate, Limiter<Void> limiter, ExecutorService executor) {
-      this.delegate = delegate;
-      this.limiter = limiter;
-      this.executor = executor;
+    ThrottledSpanConsumer(ThrottledStorageComponent throttledStorage) {
+      this.executor = throttledStorage.executor;
+      this.limiter = throttledStorage.limiter;
+      this.isOverCapacity = throttledStorage::isOverCapacity;
```

Comment:
This evaluates to creating a new Predicate.  Is that desirable given that it would just result in the `isOverCapacity` check simply having an ever increasing stack depth?